isaaclab.bat

```diff
@@ -12,17 +12,42 @@ goto main
 
 rem Helper functions
 
-rem extract the python from isaacsim
-:extract_python_exe
+rem extract Isaac Sim directory
+:extract_isaacsim_path
 rem Check if IsaacSim directory manually specified
-rem Note: for manually build isaacsim, this: _build/linux-x86_64/release
 if not "%ISAACSIM_PATH%"=="" (
     rem Use local build
-    set build_path=%ISAACSIM_PATH%
+    set isaac_path=%ISAACSIM_PATH%
 ) else (
-    rem Use TeamCity build
-    set build_path=%ISAACLAB_PATH%\_isaac_sim
+    rem check if isaacsim is installed
+    pip show isaacsim-rl > nul 2>&1
+    if errorlevel 1 (
+        rem Use TeamCity build or pip install
+        set isaac_path=%ISAACLAB_PATH%\_isaac_sim
+    ) else (
+        rem use the python executable to get the path
+        call :extract_python_exe
+        rem retrieve the isaacsim path
+        set "isaac_path="
+        for /f "delims=" %%i in ('!python_exe! -c "import isaacsim; import os; print(os.environ['ISAAC_PATH'])"') do (
+            if not defined isaac_path (
+                set "isaac_path=%%i"
+            )
+        )
+    )
+)
+rem Check if the directory exists
+if not exist "%isaac_path%" (
+    echo [ERROR] Isaac Sim directory not found at path: %isaac_path%
+    echo [ERROR] Please specify the path to Isaac Sim directory using 'ISAACSIM_PATH' environment variable.
+    exit /b 1
 )
+goto :eof
+
+rem extract the python from isaacsim
+:extract_python_exe
+rem obtain isaacsim path
+call :extract_isaacsim_path
```

Comment:
this will cause an infinite loop calling extract_isaacsim_path, which then calls extract_python_exe. should it be moved to inside the if statement?

Reply:
Done.